packages/testing/src/mocking-activity-environment.ts

```diff
@@ -54,7 +54,7 @@ export class MockActivityEnvironment extends events.EventEmitter {
   /**
    * Run a function in Activity Context
    */
-  public async run<P extends any[], R, F extends ActivityFunction<P, R>>(fn: F, ...args: P): Promise<R> {
+  public async run<F extends ActivityFunction>(fn: F, ...args: Parameters<F>): ReturnType<F> {
     return this.activity.runNoEncoding(fn as ActivityFunction<any, any>, { args, headers: {} }) as Promise<R>;
```

Comment:
This line is still using the `R` type parameter. Again, you might have no choice but to cast the return value into `Promise<Awaited<ReturnType<F>>>`.

And while you are there, can you please remove the type cast on `fn`? I very much doubt this is required.